rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -22,44 +22,32 @@
 import numpy as np
 import rocal_pybind as b
 import amd.rocal.types as types
+import ctypes
 
 class ROCALGenericImageIterator(object):
     def __init__(self, pipeline):
         self.loader = pipeline
-        self.w = b.getOutputWidth(self.loader._handle)
-        self.h = b.getOutputHeight(self.loader._handle)
-        self.n = b.getOutputImageCount(self.loader._handle)
-        color_format = b.getOutputColorFormat(self.loader._handle)
-        self.p = (1 if (color_format == int(types.GRAY)) else 3)
-        height = self.h*self.n
-        self.out_tensor = None
-        self.out_bbox = None
-        self.out_image = np.zeros((height, self.w, self.p), dtype = "uint8")
-        self.bs = pipeline._batch_size
+        self.batch_size = pipeline._batch_size
+        self.out = None
 
     def next(self):
         return self.__next__()
 
     def __next__(self):
         if(self.loader.isEmpty()):
             raise StopIteration
+        else:
+            self.output_tensor_list = self.loader.rocalGetOutputTensors()
 
-        if self.loader.run() != 0:
-            raise StopIteration
-
-        self.loader.copyImage(self.out_image)
-        if((self.loader._name == "Caffe2ReaderDetection") or (self.loader._name == "CaffeReaderDetection")):
+        self.w = self.output_tensor_list[0].batch_width() if self.w is None else self.w
+        self.h = self.output_tensor_list[0].batch_height() if self.h is None else self.h
+        self.batch_size = self.output_tensor_list[0].batch_size() if self.batch_size is None else self.batch_size
```

Comment:
Why are u getting batch size from output tensor list if u are already assigning it as Pipeline._batch_size?